apps/tangle-dapp/data/validatorDetails/useValidatorInfoCard.ts

```diff
@@ -0,0 +1,97 @@
+'use client';
+
+import { useEffect, useMemo, useState } from 'react';
+
+import useRestakingRoleLedger from '../../data/restaking/useRestakingRoleLedger';
+import useCurrentEra from '../../data/staking/useCurrentEra';
+import { getAccountInfo, getPolkadotApiPromise } from '../../utils/polkadot';
+import {
+  getProfileTypeFromRestakeRoleLedger,
+  getTotalRestakedFromRestakeRoleLedger,
+} from '../../utils/polkadot/restake';
+
+export default function useValidatorInfoCard(
+  rpcEndpoint: string,
+  validatorAddress: string
+) {
+  const { data: currentEra } = useCurrentEra();
+  const { data: ledgerOpt, isLoading: isLoadingLedgerOpt } =
+    useRestakingRoleLedger(validatorAddress);
```

Comment:
Think of creating a `RestakeContext` to share these props between the service and the restake page.

Reply:
The `RestakeContext` is for handling the data of the current user, while the service page displays the data of the whole network. I think we can keep it like this for now